### PR TITLE
AP_Scripting: add position light example

### PR DESCRIPTION
**Pull request by mirkix**
This is an example to use LUA for LED position lights. If the vehicle is disarmed the left strip will light up red the right one green. When the vehicle is armed, the LEDs will change between red / green and white flashing.

### CODE REVIEW

File: libraries/AP_Scripting/examples/LED_poslight.lua [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+--[[
+ Brightness for flash light when armed.
+--]]
+local br_flash = 255
+
+--[[
+ Use SERVOn_FUNCTION 94 for left LED strip 
+ Use SERVOn_FUNCTION 95 for right LED strip
+--]]
+local chan_left = SRV_Channels:find_channel(94)
```
Comment (IamPete1) [inline libraries/AP_Scripting/examples/LED_poslight.lua:26]:
```suggestion
local chan_left = assert(SRV_Channels:find_channel(94),"LEDs left: channel not set")
```

File: libraries/AP_Scripting/examples/LED_poslight.lua [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+-- find_channel returns 0 to 15, convert to 1 to 16
+chan_left = chan_left + 1
+chan_right = chan_right + 1
+
+gcs:send_text(6, "LEDs strip left: chan=" .. tostring(chan_left))
+gcs:send_text(6, "LEDs strip right: chan=" .. tostring(chan_right))
+
+-- initialisation code
+serialLED:set_num_neopixel(chan_left, num_leds)
```
Comment (IamPete1) [inline libraries/AP_Scripting/examples/LED_poslight.lua:47]: Probably worth also asserting these, that can return false.
```suggestion
assert(serialLED:set_num_neopixel(chan_left, num_leds),"Failed left LED setup")
```

File: libraries/AP_Scripting/examples/LED_poslight.lua [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+serialLED:set_num_neopixel(chan_right, num_leds)
+--serialLED:set_num_profiled(chan_left, num_leds)
+--serialLED:set_num_profiled(chan_right, num_leds)
+--serialLED:set_num_LEDs(chan_left, num_leds)
+--serialLED:set_num_LEDs(chan_right, num_leds)
```
Comment (IamPete1) [inline libraries/AP_Scripting/examples/LED_poslight.lua:52]: `serialLED:set_num_LEDs` is no longer valid

File: libraries/AP_Scripting/examples/LED_poslight.lua [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+function update_LEDs()
+  if arming:is_armed() then
+    if (timer == 0) then
+      serialLED:set_RGB(chan_left, -1, br_flash, br_flash, br_flash)
+      serialLED:set_RGB(chan_right, -1, br_flash, br_flash, br_flash)
+    elseif (timer == 2) then
```
Comment (IamPete1) [inline libraries/AP_Scripting/examples/LED_poslight.lua:59]: You could half your timer thresholds and run at 10hz

File: libraries/AP_Scripting/examples/LED_poslight.lua (hunk 1 above)
Comment (mirkix) [reply libraries/AP_Scripting/examples/LED_poslight.lua]: Thanks, done.

File: libraries/AP_Scripting/examples/LED_poslight.lua (hunk 2 above)
Comment (mirkix) [reply libraries/AP_Scripting/examples/LED_poslight.lua]: Thanks, done.

File: libraries/AP_Scripting/examples/LED_poslight.lua (hunk 3 above)
Comment (mirkix) [reply libraries/AP_Scripting/examples/LED_poslight.lua]: Thanks, done.

File: libraries/AP_Scripting/examples/LED_poslight.lua (hunk 4 above)
Comment (mirkix) [reply libraries/AP_Scripting/examples/LED_poslight.lua]: Thanks, done.

**Review by IamPete1 — APPROVED** [review]
LGTM, thanks

File: libraries/AP_Scripting/examples/LED_poslight.lua [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    serialLED:set_RGB(chan_left, -1, br_color, 0, 0)
+    serialLED:set_RGB(chan_right, -1, 0, br_color, 0)
+    timer = 0
+  end
+  serialLED:send(chan_left)
+  serialLED:send(chan_right)
+  return update_LEDs, 100 -- run at 10Hz
+end
+
+return update_LEDs, 1000
```
Comment (WickedShell) [inline libraries/AP_Scripting/examples/LED_poslight.lua:71]:
```suggestion
return update_LEDs()
```

This should remove the 1 second delay on starting the script

File: libraries/AP_Scripting/examples/LED_poslight.lua [hunk 6]
```diff
@@ -0,0 +1,71 @@
+--[[
+ Script to use LED strips as position lights.
```
Comment (WickedShell) [inline libraries/AP_Scripting/examples/LED_poslight.lua:2]: Not critical, but generally I'd recommend not doing block comments like this unless you are actually writing a block comment. The problem is we parse the script at runtime, so comments like this work to actually increase the RAM consumption while parsing a script, which is actually when you are most RAM constrained normally on a F4 based board.

**Review by WickedShell — COMMENTED** [review]
Not critical things, just thought I'd point them out

File: libraries/AP_Scripting/examples/LED_poslight.lua (hunk 5 above)
Comment (mirkix) [reply libraries/AP_Scripting/examples/LED_poslight.lua]: Done, thanks.

File: libraries/AP_Scripting/examples/LED_poslight.lua (hunk 6 above)
Comment (mirkix) [reply libraries/AP_Scripting/examples/LED_poslight.lua:2]: Done, thanks.

**Review by tridge — APPROVED** [review]

**Pull request merged**